packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -218,7 +219,11 @@ export function executeStepDependencyGraph<
         if (isStepEnabled(stepId) && stepDependenciesAreComplete(stepId)) {
           removeStepFromWorkingGraph(stepId);
           promiseQueue.add(() =>
-            executeStep(step).catch(handleUnexpectedError),
+            timeOperation({
+              logger: executionContext.logger,
+              metricName: `${stepId}-duration`,
```

Comment:
I wonder if this should be `duration-step`, with `dimensions: { stepId }`, so there is not an explosion of metrics and we can see averages across all steps easily within an integration.

Reply:
Seems reasonable